### DISCUSSION TITLE
Set IP whitelist for a subdomain or route

### DISCUSSION THREAD

**Discussion by mtin79** [vercel/next.js#22761] [Help]
I want to deploy a multi tenant next js app to Vercel and wonder if it's possible to only allow certain routes or subdomains of this app to be reached by users coming from certain IPs only.
As some app tenants need to be visiting the app from the companies Intranet IP range only!
Thanks for any help or sharing any ideas!

**Comment by masterkingmk** [comment]
Yes better I very soon set all of my network change plan and don’t let attack from any one in near.our want good dev for creating platform and cryptocurrency with our idea and provide backend to join kingdom sun and for all of life get income from our financial trader and crypto investor.we need 5 people very good in job not like me.give me all wrongs and just see their income.we need best.please general and best dev join us to beginning move in world.





On Thursday, March 4, 2021, 10:56 am, Martin Schubert <notifications@github.com> wrote:




I want to deploy a multi tenant next js app to Vercel and wonder if it's possible to only allow certain routes or subdomains of this app to be reached by users coming from certain IPs only.
As some app tenants need to be visiting the app from the companies Intranet IP range only!
Thanks for any help or sharing any ideas!

**Comment by misc-alt** [comment]
Was looking into this and I don't think this is possible. Do not use something like the `x-forwarded-for` header as this can be spoofed by adversaries. The only solution I could come up with is to use some sort of middleware or CDN. Cloudflare is pretty good, this documentation does just what we want:
•	Cloudflare: Require known IP addresses in site admin area
https://developers.cloudflare.com/waf/custom-rules/use-cases/site-admin-only-known-ips/
-- Aziah :)
